src/helix-ui/styles/components/checkbox.less

```diff
@@ -0,0 +1,250 @@
+  border-radius: 2px;
+  border: 1px solid @gray-500;
+  display: inline-block;
+  height: 1rem;
+  vertical-align: baseline;
+  width: 1rem;
+}
+
+/* ========== LAYOUT ========== */
+hx-checkbox-control {
+  -ms-grid-column-align: start;
+  -ms-grid-columns: auto 1fr;
+  -ms-grid-rows: auto auto;
+  display: -ms-grid;
+  display: grid;
+  grid-template-rows: auto auto;
+  grid-template-columns: auto 1fr;
+  grid-template-areas:
+    'ctrl lbl'
+    '.    help';
+  justify-items: start;
+
+  > label {
+    // grid child
+    -ms-grid-column: 2;
+    -ms-grid-row: 1;
+    grid-area: lbl;
+
+    // grid parent
+    -ms-grid-columns: auto 1fr;
+    -ms-grid-rows: auto;
+    display: -ms-grid;
+    display: grid;
+    grid-template-areas: 'facade .';
+    grid-template-columns: auto 1fr;
+    grid-template-rows: auto;
+    grid-gap: 0.5rem;
+    margin: 0;
+    padding: 0;
+
+    > hx-checkbox {
+      -ms-grid-column: 1;
+      -ms-grid-row: 1;
+      display: none;
+      grid-area: facade;
+    }
+  }
+
+  > p {
+    -ms-grid-column: 2;
+    -ms-grid-row: 2;
+    grid-area: help;
+    margin: 0;
+  }
+
+  > input[type="checkbox"] {
+    -ms-grid-column: 1;
+    -ms-grid-row: 1;
+    display: inline-block;
+    grid-area: ctrl;
+    height: 1rem;
+    margin-right: 0.50rem;
+    margin: 0;
+    width: 1rem;
+
+    &::-ms-check {
+      box-sizing: border-box;
+      display: block;
+    }
+  }
+}
+
+/* ========== PRISTINE ========== */
+hx-checkbox-control {
+  // help text
+  // TODO: centralize
+  > p {
+    color: @gray-750;
+    font-size: 0.75rem;
+  }
+
+  > input[type="checkbox"] {
+    color: @gray-500;
+
+    &::-ms-check {
+      background-color: @gray-0;
+      border-color: currentColor;
+      border-style: solid;
+      border-width: 1px;
+      color: inherit;
+    }
+
+    &:focus {
+      box-shadow: @focus-glow;
+
+      ~ label > hx-checkbox {
+        box-shadow: @focus-glow;
+      }
+    }
+
+    &:hover {
+      color: @cyan-500;
+
+      &::-ms-check {
+        background-color: @cyan-50;
+        color: @cyan-500;
+      }
+    }
+
+    &:checked {
+      ~ label > hx-checkbox {
+        --hxCheckbox__minus-display: none;
+        --hxCheckbox__tick-display: block;
+      }
+    }
+
+    &:indeterminate {
+      ~ label > hx-checkbox {
+        --hxCheckbox__minus-display: block;
+        --hxCheckbox__tick-display: none;
+      }
+    }
+
+    &:checked,
+    &:indeterminate {
+      color: @cyan-900;
+
+      ~ label > hx-checkbox {
+        border-color: @cyan-900;
+        color: @cyan-900;
+      }
+    }
+
+    &:valid {
+      // hx-checkbox-control > input[type="checkbox"]:valid >
+      ~ label > hx-checkbox:hover {
+        background-color: @cyan-50;
+        border-color: @cyan-500;
+        color: @cyan-500;
+      }
+    }
+  }
+}
+
+/* ========== INVALID ========== */
+hx-checkbox-control > input[type="checkbox"]:invalid {
+  color: @red-900;
+
+  &::-ms-check {
+    border-width: 2px;
+  }
```

Comment:
would it be useful to put the `border-color` and `color` with `!important`?

Reply:
yes! updated